hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableECContainerProvider.java

```diff
@@ -159,13 +160,21 @@ public ContainerInfo getContainer(final long size,
     // If we get here, all the pipelines we tried were no good. So try to
     // allocate a new one.
     try {
-      synchronized (this) {
-        if (openPipelineCount < maximumPipelines) {
+      if (openPipelineCount >= maximumPipelines) {
+        final int nodeCount = nodeManager.getNodeCount(null, null);
+        if (nodeCount > maximumPipelines) {
+          LOG.debug("Increasing pipeline limit {} -> {} for final attempt",
+              maximumPipelines, nodeCount);
+          maximumPipelines = nodeCount;
```

Comment:
Still users may need to tune this value based on their cluster and ECReplication config. Let's using 10:4 will different number openpipeline per node vs using 3:2 will have different number of pipelines per node. Because number of node involve in a pipeline changes. It is hard to auto tune that. (Vague thought)But later we may want to think like 80% nodes max pipeline count limit or so. if we get the info like each DN openpipelines counts, then if 80%( just an argitrary number) nodes in the cluster already have that max open pipelines, then we are at cap of openpipelines limit.